ouroboros-consensus/test-consensus/Test/Consensus/HardFork/Combinator/B.hs

```diff
@@ -0,0 +1,239 @@
+
+  protocolSecurityParam = cfgB_k
+  updateConsensusState  = \_ _ _ _ -> return ()
+  rewindConsensusState  = \_ _ _ _ -> Just ()
+
+data BlockB = BlkB {
+      blkB_header :: Header BlockB
+    , blkB_body   :: Body   BlockB
+    }
+  deriving stock    (Show, Eq, Generic)
+  deriving anyclass (NoUnexpectedThunks, Serialise)
+
+instance HasBlockBody BlockB where
+  data Body BlockB = BodyB
+    deriving stock    (Show, Eq, Generic)
+    deriving anyclass (NoUnexpectedThunks, Serialise)
+
+  getBody       = blkB_body
+  assembleBlock = Just .: BlkB
+
+instance GetHeader BlockB where
+  newtype Header BlockB = HdrB { getHdrB :: HeaderFields BlockB }
+    deriving stock    (Show, Eq, Generic)
+    deriving anyclass (NoUnexpectedThunks, Serialise)
+
+  getHeader          = blkB_header
+  blockMatchesHeader = \_ _ -> True -- We are not interested in integrity here
+  headerIsEBB        = const Nothing
+
+data instance BlockConfig BlockB = BCfgB
+  deriving (Generic, NoUnexpectedThunks)
+
+type instance BlockProtocol BlockB = ProtocolB
+type instance HeaderHash    BlockB = Hash
+
+instance ConfigSupportsNode BlockB where
+  data CodecConfig BlockB = CCfgB
+
+  getCodecConfig     _ = CCfgB
+  getSystemStart     _ = SystemStart dawnOfTime
+  getNetworkMagic    _ = NetworkMagic 0
+  getProtocolMagicId _ = ProtocolMagicId 0
+
+instance StandardHash BlockB
+
+instance Measured BlockMeasure BlockB where
+  measure = blockMeasure
+
+instance HasHeader BlockB where
+  blockHash      =            blockHash     . getHeader
+  blockPrevHash  = castHash . blockPrevHash . getHeader
+  blockSlot      =            blockSlot     . getHeader
+  blockNo        =            blockNo       . getHeader
+  blockInvariant = const True
+
+instance HasHeader (Header BlockB) where
+  blockHash      =            headerFieldHash     . getHdrB
+  blockPrevHash  = castHash . headerFieldPrevHash . getHdrB
+  blockSlot      =            headerFieldSlot     . getHdrB
+  blockNo        =            headerFieldNo       . getHdrB
+  blockInvariant = const True
+
+instance HasAnnTip BlockB where
+
+instance BasicEnvelopeValidation BlockB where
+  -- Use defaults
+
+instance ValidateEnvelope BlockB where
+
+data instance LedgerState BlockB = LgrB {
+      lgrB_tip :: Point BlockB
+    }
+  deriving (Show, Eq, Generic, NoUnexpectedThunks, Serialise)
+
+type instance LedgerCfg (LedgerState BlockB) = EraParams
+
+instance IsLedger (LedgerState BlockB) where
+  type LedgerErr (LedgerState BlockB) = Void
+  applyChainTick _ = Ticked
+
+instance ApplyBlock (LedgerState BlockB) BlockB where
+  applyLedgerBlock   = \_ b _ -> return $ LgrB (blockPoint b)
+  reapplyLedgerBlock = \_ b _ -> LgrB (blockPoint b)
+  ledgerTipPoint     = lgrB_tip
+
+instance UpdateLedger BlockB
+
+instance CanForge BlockB where
+  forgeBlock _ _ bno (Ticked sno st) _txs _ = return $ BlkB {
+      blkB_header = HdrB HeaderFields {
+          headerFieldHash     = Lazy.toStrict . B.encode $ unSlotNo sno
```

Comment:
Include "B"

Reply:
Not necessary: B will not run concurrently with A and so can't have overlapping slot numbers.